test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -728,6 +779,46 @@ public void CreateDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool
         }
 
         [Fact]
+        public void CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_NameWithIStringLocalizer()
```

Comment:
Suggest creating a small `EnumGroupAndNameTest` class and moving a few of these tests there. No need for duplication here because other tests demonstrate the `EnumGroupAndName` `struct`s are created correctly in general (not much logic left in the provider).